wled00/FX.cpp

```diff
@@ -4861,6 +4861,53 @@ static const char _data_FX_MODE_FLOWSTRIPE[] PROGMEM = "Flow Stripe@Hue speed,Ef
 #define XY(x,y) SEGMENT.XY(x,y)
 
 
+/////////////////////////////
+//    2D Analog Clock      //
+/////////////////////////////
+uint16_t mode_2DAnalogClock(void) {                  // By Andras Fekete (bandi13)
+  if (!strip.isMatrix || !SEGMENT.is2D()) return mode_static(); // not a 2D set-up
+
+  const int cols = SEGMENT.virtualWidth();
+  const int rows = SEGMENT.virtualHeight();
+  const int centerX = (cols-!(cols%2)) >> 1;  // use odd sized circle
+  const int centerY = (rows-!(rows%2)) >> 1;  // use odd sized circle
+  const int radius = min(centerX, centerY);
+  const bool soft = radius > 6 && SEGMENT.check2;
+
+  SEGMENT.fadeToBlackBy(240);
+  SEGMENT.drawCircle(centerX, centerY, radius, DARKGREY, soft);
+  if (radius > 8) SEGMENT.drawCircle(centerX, centerY, radius-1, DARKGREY, soft); // thicker circle
+
+  time_t hours_in_day = (localTime / (60 * 60)) % 12;
+  time_t minutes_in_day = (localTime / (60)) % 60;
+  time_t seconds_in_day = (localTime) % 60;
+
+  float hour_angle = radians(30.0f * (hours_in_day + minutes_in_day / 60.0f) - 90.0f);
```

Comment:
You could have left the original numbers in for code clarity. The preprocessor would have done the simplified math anyway.

Reply:
can be added into comment if you think so.
compiler is unable to optimise `360 * (hours_in_day + minutes_in_day / 60.0) / 12` as it needs to mix integer and floating point maths. with integer maths precedence is important as you are loosing precision when dividing before multiplication.
that may not hold true with every compiler though.

Reply:
Why not go full Monty and do `radians(30.0f * hours_in_day + minutes_in_day / 2.0f - 90.0f)`? Divide by 2 is more efficient than another number (just decrement the mantissa IIRC).

A comment is definitely warranted. Though we could go back to the original equation and have all the constants be `float` which should let the preprocessor figure out that it needs to do `360 / 12`.

Reply:
> Why not go full Monty and do `radians(30.0f * hours_in_day + minutes_in_day / 2.0f - 90.0f)`? Divide by 2 is more efficient than another number (just decrement the mantissa IIRC).

You can do that if you want. Anything that reduces CPU clock cycles is welcome.